docs/extending-minerva.rst

```diff
@@ -0,0 +1,68 @@
+Extending Minerva
+=================
+
+Creating a Minerva Plugin
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Minerva utilizes the Girder plugin system, so it will be worthwhile to familiarize yourself with their section on `Plugin Development <http://girder.readthedocs.org/en/latest/plugin-development.html>`_.
+
+For Minerva specific plugins, specifying "minerva" as a dependency within your plugin is required to ensure things load in the proper order. Take this example plugin configuration:
+
+.. code-block:: python
+
+   {
+       "name": "My Minerva Plugin",
+       "dependencies": ["minerva"]
+   }
+
+You should have a Minerva plugin that does nothing, and `can be enabled through the administration console <http://girder.readthedocs.org/en/latest/installation.html#initial-setup>`_. 
+
+
+Extending the look and feel of Minerva
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+While Minerva plugins follow the guidelines provided by Girder's client side, there are additional concerns when writing for Minerva.
+
+.. note:: Girder has documentation on `extending the client-side application <http://girder.readthedocs.org/en/latest/plugin-development.html#extending-the-client-side-application>`_ which also applies to Minerva.
+
+Minerva Panel Views
+-------------------
+Minerva renders panels in a particular way, as a result there are some guidelines that need be followed when creating your own Panels:
+
+- Panels should never call render within their initialize function
+- Panel views must extend the base Panel view
+- Panel views need to extend their events to take advantage of collapsible, removable, configurable panels
+
+
+Configure Minerva's layout
+--------------------------
+Taking our example plugin from before, we can alter how Minerva displays different panels.
+
+Let's pretend our use case of Minerva deems the Jobs Panel useless, and is more focused on the datasets available. In this case one might want to disable the Jobs Panel entirely, and move the datasets panel to the top.
+
+From your plugin root, create a JavaScript file at ``web_client/js/some-file.js`` and add the following code:
+
+.. code-block:: javascript
+
+   girder.events.once('m:pre-render-panel-groups', function (sessionView) {
+       var leftPanelGroup = _.find(sessionView.layout.panelGroups, function (panelGroup) {
+           return panelGroup.id === 'm-left-panel-group';
+       });
+
+       // Remove the jobs panel
+       leftPanelGroup.panelViews = _.filter(leftPanelGroup.panelViews, function (panelView) {
```

Comment:
I want to steer clear of the term "hide" because if anyone wants to hide anything they should be using CSS. Everything I'm doing actually *disables* the panel.

The relationship here is that ```removePanel``` is an event which disables a specific panel and then removes the DOM element completely, while this code is hooking in to disable a panel before it is ever rendered.

I will add a method to the session view to disable a given panel and utilize that in removePanel and the documentation.